src/source_sampling.cpp

```diff
@@ -85,7 +88,8 @@ pyne::Sampler::Sampler(std::string filename,
                  int mode)
   : filename(filename),
     tag_names(tag_names),
-    e_bounds(e_bounds) {
+    e_bounds(e_bounds),
+    mode(mode) {
   // determine the bias_mode
   if (mode == 0){
```

Comment:
On the C/C++ side it would probably be best to continue using the `MeshMode` enumerators where possible.

Reply:
This mode is passed from Fortran. DAGMC input -> source.F90 -> source_sampling.cpp. The mode could be (0, 1, 2, 4, 5, 6). We have to keep it as an input parameter.
We can't assign `mesh_mode` from this mode, because the `TET` mesh and `VOXEL` share the modes (0, 1, 2). We need extra checks when we get both the `ve_type` and `mode`.